### PR TITLE
Adding steps for how to use existing cluster as bootstrap cluster.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/825

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Adding steps for how to use existing cluster as bootstrap cluster.
```

### CODE REVIEW

**Comment by jichenjc** [comment]
I tried this command with existing minikube env on openstack provider
I found some bug though but it's related to openstack, the command for existing cluster does work

File: cmd/clusterctl/README.md [hunk 1]
```diff
@@ -40,13 +43,21 @@ https://github.com/kubernetes-sigs/cluster-api/issues/158 and https://github.com
...
    ```shell
-   ./clusterctl create cluster --provider <provider> --bootstrap-type <bootstrap-type> -c cluster.yaml -m machines.yaml -p provider-components.yaml -a addons.yaml
+   ./clusterctl create cluster --provider <provider> --bootstrap-type <bootstrap-type> -c cluster.yaml \
+     -m machines.yaml -p provider-components.yaml -a addons.yaml
    ```
 
-Currently two `bootstrap-type` options are supported - `kind` and `minikube`.
+   If you are using minikube, to choose a specific minikube driver, please use the `--bootstrap-flags vm-driver=xxx` command line parameter. For example to use the kvm2 driver with clusterctl you woud add `--bootstrap-flags vm-driver=kvm2`.
+
+   - With `bootstrap-cluster-kubeconfig`, this is used when you are using an existing Kubernetes cluster as your bootstrap cluster.
```
Comment (vincepri) [inline cmd/clusterctl/README.md]:
```suggestion
   -  __Existing Cluster__:  Use `bootstrap-cluster-kubeconfig`. This flag is used when you have an existing Kubernetes cluster.
```

File: cmd/clusterctl/README.md [hunk 2]
```diff
@@ -13,8 +13,11 @@ this repository.**
 
 ### Prerequisites
 
-1. Install [kind](https://github.com/kubernetes-sigs/kind#installation-and-usage) or [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/). [kind](https://github.com/kubernetes-sigs/kind#installation-and-usage) is preferred.
-2. If you are using kind, go to step 3; If you are using minikube, install a [driver](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md) for minikube. For Linux, we recommend kvm2. For MacOS, we recommend VirtualBox.
+1. You can use either VM, container or existing Kubernetes cluster act as bootstrap cluster, it is recommended to use container or existing Kubernetes cluster.
```
Comment (vincepri) [inline cmd/clusterctl/README.md]:
```suggestion
1. Cluster API runs its operations in Kubernetes. A pre-existing or temporary bootstrap cluster is required. Currently, we support multiple methods to bootstrap Cluster API: `kind` (preferred), `minikube` or any pre-existing cluster.
```

File: cmd/clusterctl/README.md [hunk 3]
```diff
@@ -13,8 +13,11 @@ this repository.**
...
+   - If you want to use VM, install [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), version 0.30.0 or greater.
+   - If you want to use container, install [kind](https://github.com/kubernetes-sigs/kind#installation-and-usage).
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply cmd/clusterctl/README.md]: Done

**Comment by detiber** [comment]
+1 from me

**Pull request merged**